virttest/iscsi.py

```diff
@@ -347,6 +347,26 @@ def get_device_name(self):
             LOG.error("Session is not logged in yet.")
         return device_name
 
+    def get_devices(self):
+        """
+        Get devices from the target name.
+        """
+        cmd = "iscsiadm -m session -P 3"
+        pattern = r"Lun:\s(\d+).*?scsi\sdisk\s(\w+)\s+State:\srunning"
```

Comment:
the pattern doesn't work well for the following:
Attached SCSI devices:
************************
Host Number: 17 State: running
scsi17 Channel 00 Id 0 Lun: 0
Attached scsi disk sdf          State: running
scsi17 Channel 00 Id 0 Lun: 1
Attached scsi disk sdi          State: ing         >>>>>>not running here
scsi17 Channel 00 Id 0 Lun: 2
Attached scsi disk sdh          State: running
scsi17 Channel 00 Id 0 Lun: 3
Attached scsi disk sdg          State: running

I cannot find an easy way to handle this, just fyi.
pattern = r"Lun:\s(\d+).*?scsi\sdisk\s(\w+)\s+State:\s(\w+)"
devices = {t[0]:t[1] for t in re.findall(pattern, c, re.S) if t and t[2] == 'running'}